### PR TITLE
Remove obsolete DWARF Archetype mangling

### PR DESCRIPTION
**Pull request by adrian-prantl**

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test and merge

File: docs/ABI/Mangling.rst [hunk 1]
```diff
@@ -467,7 +467,6 @@ Types
 
   assoc-type-list ::= assoc-type-name '_' assoc-type-name*
 
-  archetype ::= context 'Qq' INDEX           // archetype+context (DWARF only)
   archetype ::= associated-type
```
Comment (slavapestov) [inline docs/ABI/Mangling.rst]: Can you remove this line too?

**Review by slavapestov — APPROVED** [review]
Excellent, thanks! You can remove some code from type reconstruction now too right?

**Comment by slavapestov** [comment]
I think it’s worth checking the demangler and remangler too.

File: docs/ABI/Mangling.rst (hunk 1 above)
Comment (adrian-prantl) [reply docs/ABI/Mangling.rst]: which one?

**Comment by adrian-prantl** [comment]
Do we want to remove the demangler support or is there value in being able to demangle older typenames?

File: docs/ABI/Mangling.rst (hunk 1 above)
Comment (slavapestov) [reply docs/ABI/Mangling.rst]: archetype ::== associated-type

There’s no archetype production any longer

**Comment by slavapestov** [comment]
I don’t think we care because it’s a dwarf-only mangling

**Comment by DougGregor** [comment]
Yaaaaaaaay

**Pull request merged**